zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -495,11 +508,12 @@ impl<F: Field> MPTConfig<F> {
                         )?;
                         cached_region.pop_region();
                     }
-
-                    //println!("stored expressions");
-                    //println!("challenge accessed: {:?}", *cached_region.challenges().indexed()[2]);
+                    
+                    // Region for spliting dynamic lookup
+                    // currently not working :(
```

Comment:
Yeah this is gonna by a pain to get working now because `build_dynamic_lookups` does the splitting outside of any region, and so could use values of different regions.

But instead of doing things inside `build_dynamic_lookups`, why not do it in `add_lookup`? That seems to be the easier place and don't have these problems.